lib/peeps.rb

```diff
@@ -0,0 +1,54 @@
+require 'pg'
+
+class Peeps
+
+  attr_reader :message, :date
+
+  def initialize(message:, date:)
+    @message = message
+    @date = date
+  end
+
+  def self.all
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec('SELECT * FROM peeps ORDER BY id DESC;')
+    
+    result.map do |peep| 
+      Peeps.new(message: peep['message'], date: peep['date'])
+    end
+  end
+
+  def self.add(peep)
+    date = Time.new
+    store_date = "#{date.hour}:#{date.min}, #{date.day}.#{date.month}.#{date.year}"
+
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    result = connection.exec("INSERT INTO peeps (message, date) values ('#{peep}', '#{store_date}');")
```

Comment:
Here, there are broadly two options in terms of how to achieve the search feature.

1. Retrieve all the peeps from the DB and then filter them in the controller or model, using Ruby.
2. Use SQL to only return the peeps of interest from the DB (which is the option you chose).

I'd say you made a good choice - nice work! Do you know why the one you chose is the better of these two options?